lib/Basics/error.cpp

```diff
@@ -45,22 +54,44 @@ thread_local ErrorContainer LastError;
 ErrorCode TRI_errno() { return LastError._number; }
 
 /// @brief returns the last error as string
-std::string_view TRI_last_error() {
+std::string TRI_last_error() {
   ErrorCode err = LastError._number;
 
   if (err == TRI_ERROR_SYS_ERROR) {
+#if defined(_WIN32)
+    if (LastError._sys == 0) {
+      return {};
+    }
+    LPSTR messageBuffer = nullptr;
+    std::size_t size = FormatMessageA(
+        FORMAT_MESSAGE_ALLOCATE_BUFFER | FORMAT_MESSAGE_FROM_SYSTEM |
+            FORMAT_MESSAGE_IGNORE_INSERTS,
+        nullptr, LastError._sys, 0, (LPSTR)&messageBuffer, 0, nullptr);
+    if (size == 0) {
+      return "failed to get error message, the error code is " +
+             std::to_string(LastError._sys);
+    }
+    std::string message(messageBuffer, size);
+    LocalFree(messageBuffer);
```

Comment:
This is not exception-safe, i.e. if `std::string(messageBuffer, size);` throws due to `std::bad_alloc`, the messageBuffer allocation is leaked. Would be better to free it using a scopeGuard or another RAII mechanism that is guaranteed to be executed.

Reply:
Thanks for pointing that out! I have revised the code to include a scope guard. We should be safe now, even if an exception is thrown. Please take a look at the updated changes and let me know if you have further feedback or suggestions.